src/client/widgets/ConfirmationModal.js

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react';
+import { Modal, message } from 'antd';
+import { upperFirst } from 'lodash';
+import { injectIntl } from 'react-intl';
+import { withRouter } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { estimateAmount, finalConfirmation } from '../../waivioApi/ApiClient';
+import { CRYPTO_FOR_VALIDATE_WALLET } from '../../common/constants/waivio';
+import { isGuestUser } from '../reducers';
+
+const ConfirmationModal = ({ intl, history }) => {
+  const [count, calculateAmount] = useState(0);
+  const [isVisible, setVisible] = useState(true);
+  const [isLoading, setLoading] = useState(false);
+  const isGuest = useSelector(isGuestUser);
+  const parseSearchParams = history.location.search
+    .split('&')
+    .map(search => search.replace('=', '": "').replace('?', '"'))
+    .join('", "');
+  const searchParams = JSON.parse(`{${parseSearchParams}"}`);
```

Comment:
you can get query string params using **URLSearchParams** (https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)